sdk/python/lib/pulumi/output.py

```diff
@@ -321,49 +322,71 @@ def secret(val: Input[T]) -> 'Output[T]':
         return Output(o._resources, o._future, o._is_known, is_secret)
 
     @staticmethod
-    def all(*args: Input[T]) -> 'Output[List[T]]':
+    def all(*args: Input[T], **kwargs: Input[T]) -> 'Union[Output[List[T]], Output[Dict[T]]]':
```

Comment:
I was thinking we'd use [`@typing.overload`](https://docs.python.org/3/library/typing.html#typing.overload) for this, something like:

```python
@overload
@staticmethod
def all(*args: Input[T]) -> 'Output[List[T]]':
    ...

@overload
@staticmethod
def all(**kwargs: Input[T]) -> 'Output[Dict[T]]':
    ...

@staticmethod
def all(*args: Input[T], **kwargs: Input[T]):
    # implementation
```

Reply:
TIL! Thanks, will fix up.